dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/UserHelper.cs

```diff
@@ -237,4 +242,42 @@ public async Task<IActionResult> GeneratePinForExistingUserAccount(SignInJourney
 
         await _dbContext.SaveChangesAsync();
     }
+
+    public async Task EnsureDqtUserNameMatch(User user, AuthenticationState authenticationState)
+    {
+        var dqtUser = await _dqtApiClient.GetTeacherByTrn(user.Trn!);
+
+        if (dqtUser is not null &&
+            NameHelper.GetFullName(user.FirstName, user.MiddleName, user.LastName) !=
+            NameHelper.GetFullName(dqtUser.FirstName, dqtUser.MiddleName, dqtUser.LastName))
+        {
+            await AssignDqtUserName(user.UserId, dqtUser);
+            authenticationState.OnNameSet(dqtUser.FirstName, dqtUser.MiddleName, dqtUser.LastName);
+        }
+    }
+
+    private async Task AssignDqtUserName(Guid userId, TeacherInfo dqtUser)
+    {
+        var existingUser = await _dbContext.Users.SingleAsync(u => u.UserId == userId);
+
+        var changes = (existingUser.FirstName != dqtUser.FirstName ? Events.UserUpdatedEventChanges.FirstName : 0) |
+                      (existingUser.MiddleName != dqtUser.MiddleName ? Events.UserUpdatedEventChanges.MiddleName : 0) |
+                      (existingUser.LastName != dqtUser.LastName ? Events.UserUpdatedEventChanges.LastName : 0);
+
+        existingUser.FirstName = dqtUser.FirstName;
+        existingUser.MiddleName = dqtUser.MiddleName;
+        existingUser.LastName = dqtUser.LastName;
+
+        _dbContext.AddEvent(new Events.UserUpdatedEvent()
+        {
+            Source = Events.UserUpdatedEventSource.TrnMatchedToExistingUser,
```

Comment:
Don't really think this is the correct source to use, perhaps need to add another?

Reply:
this can now be changed to UserUpdatedEventSource.DqtSynchronization

Reply:
Sorted